core/evm.go

```diff
@@ -63,14 +63,17 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 	}
 
 	// Set goldTokenAddress
-	caller := &CeloBackend{config, statedb.Copy()}
-	registry, err := abigen.NewRegistryCaller(contracts_config.RegistrySmartContractAddress, caller)
-	if err != nil {
-		log.Error("Failed to access registry!", "err", err)
-	}
-	goldTokenAddress, err := registry.GetAddressForOrDie(&bind.CallOpts{}, contracts_config.GoldTokenRegistryId)
-	if err != nil {
-		log.Error("Failed to get address for GoldToken!", "err", err)
+	var goldTokenAddress common.Address
+	if config.IsCel2(header.Time) {
```

Comment:
curious... we interpret that is' cel2 using the header.time? isn't a block number?

Reply:
Yes, after the merge, hard forks are defined by time instead of block number. E.g. [ShanghaiTime is 1681338455](https://github.com/celo-org/op-geth/blob/0c10250b1e8a67815b85e59a9b3502e94b5368a5/params/config.go#L73).